### ISSUE TITLE
Train and test validation splitting

### ISSUE THREAD

**Issue by iSach** [jx-wang-s-group/CoNFiLD#9]
Hello,

While trying to reproduce your results, I had issues with my diffusion model performing poorly on the checkpoint you provide for case 2. The latents I used for training the diffusion model are the "hidden_states" of shape (1200, 256) provided in the checkpoint `checkpoint_44500.ckpt` in your Zenodo files. I converted, according to the paper, into 945 trajectories of shape (256, 256), then split this into 80/20%. Although this practice is already questionable, due to data overlap, I still converted into train/test AFTER converting to trajectories, by selecting the 756 first ones for training and the other next ones for validation. The data overlap is because the last training states are included in the first validation trajectories. 

The resulting diffusion model would perform very poorly, no matter the configuration, which led me to try another data. I tried directly the data you provide in `train_data.npy` and `valid_data.npy`, which is already split and converted to trajectories. The diffusion model reaches a much lower test loss on this data. When comparing with my splitting method, I noticed you do not split as I did.

You seem to randomly select 80% of the 945 trajectories to build your training set. Considering the U-Net "invariance" to offsets, this basically means your training and validation splits are more or less the same data as trajectories overlap so much! Can you confirm this splitting method? This is not a proper splitting method and it seems ConFiLD does *not* generalize well over its learned latent space. As this is done, the data is contaminated and test loss does not mean anything. It seems the model does not generalize at all and overfits the distribution of the encoded latent codes.

Thanks in advance,
Sacha Lewin.

**Comment by meetcfd** [comment]
Hi,

Thanks for the question. The problem you are facing is a known problem of memorization within diffusion problems.

To mitigate this, we have recently proposed Patched ConFiLD architecture, which has shown strong generalization for data assimilation  for unseen timesteps and unseen flow parameters. You can read the preprint [here](https://arxiv.org/abs/2511.12455).

Hope this helps

**Comment by iSach** [comment]
Hello,

This is not memorization. I am commenting about the overlap between your training and test splits. No architecture will fix this.
